Realm/Realm.Weaver/Analytics/Analytics.cs

```diff
@@ -352,13 +354,15 @@ private void AnalyzeClassMethods(TypeDefinition type)
                         // add_RealmChanged
                         // add_PropertyChanged
                         // get_DynamicApi
-                        key = key.Substring(index + 4);
+                        key = key.Substring(prefix.Length);
                     }
 
-                    if (!_apiAnalysisSetters.ContainsKey(key) && cil.Operand is MethodReference methodReference &&
+                    if (!_apiAnalysisSetters.ContainsKey(key) &&
+                        cil.Operand is MethodReference methodReference &&
                         methodReference.ReturnType.DeclaringType != null)
                     {
                         // when dealing with ThreadSafeReference
+                        // TODO nikola: why?
```

Comment:
Per comment - not sure I understand why we're special casing TSR like this?

Reply:
This should have simply been
`key = methodReference.ReturnType.DeclaringType.Name;`
I shouldn't have done `.ToString()`. It was at the beginning when I was not as comfortable moving around in Mono.Cecil.